class.jetpack.php

```diff
@@ -350,10 +350,89 @@ function plugin_upgrade() {
 				}
 
 				Jetpack::maybe_set_version_option();
+
+				add_action( 'wp_loaded', array( __CLASS__, 'migrate_widget_visibility' ) );
 			}
 		}
 	}
 
+	/**
+	 * Upgrade routine to go through all widgets and move the Post Type
+	 * setting to its newer location.
+	 *
+	 * @since 4.7.1
+	 *
+	 */
+	static function migrate_widget_visibility() {
```

Comment:
@zinigor should we move this method over to somewhere in the `Jetpack_Widget_Conditions` class so that we can use it when we merge these changes to wpcom?